### DISCUSSION TITLE
Google maps API agora é pago?

### DISCUSSION THREAD

**Discussion by leandroruel** [frontendbr/forum#1238] [Discussão]
Estava trabalhando em um formulário de contato em um site quando derrepente, me aparece uma mensagem de erro no console, após ter inserido um mapa do google na página, me pareceu estranho, mas fazia um bom tempo que não usava o mapa do google, verifiquei as mensagens e descobri que a api do google maps agora é pago, e em ordem para utilizá-lo em seu site, você deve se cadastrar em uma conta do google e obter uma chave de api, e deixar suas informações de cobrança (cartão de crédito) no site, para uma eventual cobrança pelo mapa, se vocês já passaram por esse problema, estão usando outra solução?

lembrando que a opção de embedar aquele `<iframe>` ainda é liberada, mas agora quebrou todos os plugins de mapa.

https://www.logicdesign.co.uk/2018/09/google-maps-api-now-paid-service/

**Comment by tcelestino** [comment]
> The new pricing model is a freemium one – all users will receive $200 worth of API calls free every month. For the majority of websites, this monthly credit is more than enough to support the needs of a simple map on the contact page, or store finder type functionality

Acho que no seu caso é bem isso, não deve cobrar. 

Existem outras soluções:

- [HereMap](https://developer.here.com)
- [OpenStreetMap](https://www.openstreetmap.org/#map=2/-3.2/-61.2)
- [Leaft](https://leafletjs.com)

**Comment by mksanti** [comment]
Se não estou enganado sempre foi pago, mas você têm um limite de trafego free que geralmente para sites institucionais nunca atinge.

**Comment by includebrasil** [comment]
Era FREE porém de um tempo para cá o Google passou a cobrar pela API.

Eu uso em vários sites e não tenho problemas porém se ultrapassar 25 mil
visualizações vc deve pagar pelo uso.

Atenciosamente,
Ricardo Lobo (Webdesign)
--------------------------------------------------
Telefones:

+ 55 [73] 3633-6425
+ 55 [73] 8815-7666
--------------------------------------------------
E-mail: ricardo@nitrodesign.com.br
Website: http://www.nitrodesign.com.br


Em dom, 14 de out de 2018 às 14:41, Michael Santillán <
notifications@github.com> escreveu:

**Comment by faelribeiro22** [comment]
Tem uma calculadora com os custos da utilização dos serviços do google maps. A embed api é grátis, o resto é pago. Porém eles oferecem 200 dólares por mês para utilizar de forma gratuita. De uma olhada nesse site que tem essas informações (a calculadora fica abaixo do botão primeiros passos): https://cloud.google.com/maps-platform/pricing/?hl=pt-BR

**Comment by victornery** [comment]
Me deparei com esse mesmo problema a uns dias atrás, antes não tinha essa limitação. Existem algumas outras soluções gratuitas.

**Comment by waghcwb** [comment]
Outra alternativa seria o mapbox
https://www.mapbox.com

**Comment by ghost** [comment]
Me corrijam se estiver falando besteira mas no caso o erro não seria da aplicação que requisita este APP ? Entenda, se você tem um formulário de contato e exibe um simples mapa deveria fazer cache desta solicitação, já que é sempre a mesma, assim não utiliza esse 25.000 de limite.

**Comment by Alucabr** [comment]
> Outra alternativa seria o mapbox
> https://www.mapbox.com

mas vc não tem que criar do zero ou comprar um mapa pra usar o mapbox?

**Comment by viralatron** [comment]
Essa eu não sabia. Mas eu já achava estranho a galera da Niantic ter migrado do GMaps para o OpenStreetMaps para o Pokémon GO... deve ter sido por isso.

https://www.openstreetmap.org/

**Comment by iaurg** [comment]
O Google Maps sempre foi pago, após x requisições eles cobram. Porém agora exigem chave de API para algumas utilizações.

Se você for direto pelo link que acha no Google para gerar a chave de API eles vão te "forçar" a cadastrar os dados do cartão.

Mas caso você vá direto pelo link de developer dentro do painel no Google, consegue gerar a chave: [https://console.cloud.google.com/google/maps-apis/apis/maps-embed-backend.googleapis.com](https://console.cloud.google.com/google/maps-apis/apis/maps-embed-backend.googleapis.com)

Acesse o link acima e depois:

1. Cria um novo projeto
2. Preenche as informações de seu site/dominio -> Criar 
3. Selecione seu novo projeto 
4. Escolhe Maps JavaScript API ou Maps Embed API 
5. Gerenciar -> Credenciais -> Credenciais em APIs e serviços -> Criar credenciais -> Chave API 

E pronto, você pode usar essa chave para colocar o embed em seu site. Caso eles precisem cobrar você futuramente entrarão em contato, para ai sim você cadastrar seus dados de cobrança.

**Comment by rickson-simoes** [comment]
Agora aparentemente ficou mais burocrático ainda, pois está tendo necessidade de cartão de crédito para dar autenticidade na chave.
